code/modules/projectiles/projectile/beams.dm

```diff
@@ -153,3 +153,48 @@
 	tracer_type = /obj/effect/projectile/laser_omni/tracer
 	impact_type = /obj/effect/projectile/laser_omni/impact
 
+/obj/item/projectile/beam/mousegun
+	name = "diffuse electrical arc"
+	icon_state = "stun"
+	nodamage = 1
+	damage_type = HALLOSS
+
+	muzzle_type = /obj/effect/projectile/stun/muzzle
+	tracer_type = /obj/effect/projectile/stun/tracer
+	impact_type = /obj/effect/projectile/stun/impact
+
+/obj/item/projectile/beam/mousegun/on_impact(var/atom/A)
+	mousepulse(A, 1)
+	..()
+
+/obj/item/projectile/beam/mousegun/proc/mousepulse(turf/epicenter, range, log=0)
+	if(!epicenter) return
+
+	if(!istype(epicenter, /turf))
+		epicenter = get_turf(epicenter.loc)
+
+	for(var/atom/T in range(range, epicenter))
+		var/distance = get_dist(epicenter, T)
+		if(distance < 0)
+			distance = 0
+		if(distance <= range)
+			var/mob/M = T
+			if(istype(M, /mob/living/simple_animal/mouse) || istype(M, /mob/living/simple_animal/lizard) || istype(M, /mob/living/simple_animal/mushroom || istype(M, /mob/living/simple_animal/chick) || istype(M, /mob/living/simple_animal/corgi/puppy) || istype(M, /mob/living/simple_animal/cat/kitten)))

```

Comment:
I will not base it on mob size, because size is not the operative parameter. Pest-level is. Syndicate manhacks and drones are not pests, whereas chicks and mice and kittens are.